pkg/controller/kubebuilderx/controller.go

```diff
@@ -90,22 +92,24 @@ func (c *controller) Do(reconcilers ...Reconciler) Controller {
 }
 
 func (c *controller) Commit() error {
+	c.logger.Info("**************** commit start")
 	if c.err != nil {
 		return c.err
 	}
 	if c.oldTree.GetRoot() == nil {
 		return nil
 	}
-	builder := NewPlanBuilder(c.ctx, c.cli, c.recorder, c.logger)
+	builder := NewPlanBuilder(c.ctx, c.cli, c.oldTree, c.tree, c.recorder, c.logger)
 	if err := builder.Init(); err != nil {
 		return err
 	}
-	builder.AddTransformer(newObjectTree2DAGTransformer(c.oldTree, c.tree))
 	plan, err := builder.Build()
 	if err != nil {
 		return err
 	}
-	return plan.Execute()
+	err = plan.Execute()
+	c.logger.Info("**************** commit done")
```

Comment:
log the err result?

Reply:
not necessary. this log is for plan execution duration calculation, the error will be log by the controller-runtime.